warehouse/integrations/postgres/load.go

```diff
@@ -0,0 +1,676 @@
+	deleteDedup              = "dedup_deletion"
+	insertDedup              = "dedup_insertion"
+	dedupStage               = "dedup_stage"
+)
+
+type LoadTable struct {
+	Logger             logger.Logger
+	DB                 *sql.DB
+	Namespace          string
+	Warehouse          *warehouseutils.Warehouse
+	Stats              stats.Stats
+	Config             *config.Config
+	LoadFileDownloader load_file_downloader.LoadFileDownloader
+}
+
+type LoadUsersTable struct {
+	Logger             logger.Logger
+	DB                 *sql.DB
+	Namespace          string
+	Warehouse          *warehouseutils.Warehouse
+	Stats              stats.Stats
+	Config             *config.Config
+	LoadFileDownloader load_file_downloader.LoadFileDownloader
+}
+
+func (lt *LoadTable) Load(ctx context.Context, tableName string, tableSchemaInUpload warehouseutils.TableSchemaT) (string, error) {
+	var (
+		err                     error
+		query                   string
+		stagingTableName        string
+		txn                     *sql.Tx
+		result                  sql.Result
+		stmt                    *sql.Stmt
+		gzFile                  *os.File
+		gzReader                *gzip.Reader
+		csvRowsProcessedCount   int
+		rowsAffected            int64
+		stage                   string
+		csvReader               *csv.Reader
+		loadFiles               []string
+		sortedColumnKeys        []string
+		diagnostic              Diagnostic
+		skipDedupDestinationIDs []string
+	)
+
+	skipDedupDestinationIDs = lt.Config.GetStringSlice("Warehouse.postgres.skipDedupDestinationIDs", nil)
+	diagnostic = Diagnostic{
+		Logger:    lt.Logger,
+		Stats:     lt.Stats,
+		Config:    lt.Config,
+		Namespace: lt.Namespace,
+		Warehouse: lt.Warehouse,
+	}
+
+	query = fmt.Sprintf(`SET search_path TO %q`, lt.Namespace)
+	if _, err = lt.DB.ExecContext(ctx, query); err != nil {
+		return "", fmt.Errorf("setting search path: %w", err)
+	}
+
+	lt.Logger.Infow("started loading",
+		logfield.SourceID, lt.Warehouse.Source.ID,
+		logfield.SourceType, lt.Warehouse.Source.SourceDefinition.Name,
+		logfield.DestinationID, lt.Warehouse.Destination.ID,
+		logfield.DestinationType, lt.Warehouse.Destination.DestinationDefinition.Name,
+		logfield.WorkspaceID, lt.Warehouse.WorkspaceID,
+		logfield.Namespace, lt.Namespace,
+		logfield.TableName, tableName,
+	)
+
+	loadFiles, err = lt.LoadFileDownloader.Download(ctx, tableName)
+	defer misc.RemoveFilePaths(loadFiles...)
+	if err != nil {
+		return "", fmt.Errorf("downloading load files: %w", err)
+	}
+
+	txn, err = lt.DB.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return "", fmt.Errorf("beginning transaction: %w", err)
+	}
+
+	defer func() {
+		if err != nil {
+			diagnostic.TxnRollback(txn, tableName, stage)
+		}
+	}()
+
+	// Creating staging table
+	sortedColumnKeys = warehouseutils.SortColumnKeysFromColumnMap(tableSchemaInUpload)
+	stagingTableName = warehouseutils.StagingTableName(provider, tableName, tableNameLimit)
```

Comment:
Flag for temporary table.

Reply:
We are now having legacy implementation back driven by config `Warehouse.postgres.useLegacy` set to true by default.